File: spoj/1112-NSTEPS/solution_1112.py

def n_step(x, y):
    total = x + y
    condition = 2 >= x - y >= 0 and total % 2 == 0

    if x % 2 == 0 and condition:
        return total
    if x % 2 != 0 and condition:
        return total - 1
    return -1

if __name__ == "__main__":
    n = int(input())

    for x in range(n):
        inp = input().split(" ")
        x, y = list(map(int, inp))

        val = n_step(x, y)
        print(("No Number", val)[val != -1])

File: spoj/1112-NSTEPS/solution_1112.cpp
#include <iostream>

using namespace std;

int numberStep(int x, int y);

int main()
{
    int n;
    cin >> n;

    for (int i = 0; i < n; i++)
    {
        int x, y;
        cin >> x >> y;
        int val = numberStep(x, y);
        if (val == -1)
            cout << "No Number";
        else
            cout << val;
        cout << "\n";
    }
    return 0;
}

int numberStep(int x, int y)
{
    int val = x + y;
    bool condition = (x - y) <= 2 && x >= y;

    if (x % 2 == 0 && y % 2 == 0 && condition)
        return val;
    if (x % 2 != 0 && y % 2 != 0 && condition)
        return val - 1;
    return -1;
}